notifications/bridge.go

```diff
@@ -108,7 +109,12 @@ func (b *bridge) createManifestEvent(action string, repo string, sm distribution
 	event.Target.Size = int64(len(p))
 	event.Target.Digest = digest.FromBytes(p)
 
-	event.Target.URL, err = b.ub.BuildManifestURL(repo, event.Target.Digest.String())
+	ref, err := reference.WithDigest(repo, event.Target.Digest)
```

Comment:
It should not error if the digest passed in is valid. But the type system doesn't strictly enforce this - for example, someone could call `reference.WithDigest(ref, digest.Digest("something invalid"))`